apps/api/R/workflows.R

```diff
@@ -229,3 +244,82 @@ getWorkflowFile <- function(req, id, filename, res){
     return(bin)
   }
 }
+
+
+#################################################################################################
+#' Get the list of files in a workflow specified by the id
+#' @param id Workflow id (character)
+#' @return List of files
+#' @author Nihar Sanda
+#* @serializer contentType list(type="application/octet-stream")
+#* @get /<id>/files
+
+getWorkflowFileDetails <- function(req, id, res){
+  Workflow <- tbl(global_db_pool, "workflows") %>%
+    select(id, user_id) %>%
+    filter(id == !!id)
+  
+  qry_res <- Workflow %>% collect()
+  
+  if (nrow(qry_res) == 0) {
+    res$status <- 404
+    return(list(error="Workflow with specified ID was not found"))
+  }
+  else {
+    file_names <- list()
+    file_names <- list.files(paste0(Sys.getenv("DATA_DIR", "/data/"), "workflows/PEcAn_", id))
+     
+    return(list(workflow_id = id))
+  }
+}
+
+#################################################################################################
+#' Get the zip of specified files of the workflow specified by the id
+#' @param id Workflow id (character)
+#' @return Details of requested workflow
+#' @author Nihar Sanda
+#* @serializer contentType list(type="application/octet-stream")
+#* @post /<id>/file-multiple/
+
+getWorkflowFilesAsZip <- function(req, id, filenames, res){
+  if(req$HTTP_CONTENT_TYPE == "application/json") {
+    filenames_req <- req$postBody
+  }
+  
+  filenamesList <- jsonlite::fromJSON(filenames_req)
+  filenames <- filenamesList$files
+  
+  Workflow <- tbl(global_db_pool, "workflows") %>%
+    select(id, user_id) %>%
+    filter(id == !!id)
+  
+  qry_res <- Workflow %>% collect()
+  
+  if (nrow(qry_res) == 0) {
+    res$status <- 404
+    return()
+  }
+  else {
+    full_files <- vector(mode = "character", length = length(filenames))
+    for (i in 1:length(filenames)) {
+      
+      # Check if the requested file exists on the host
```

Comment:
If you're doing this check both here and in the last function then it seems like a good place to create a function that does the check. That way you only need to fix errors in one place? This code has the same filepath error as the previous and neither seems to be checkin whether the workflow being queried is on the local machine or not

Reply:
Sure sir, Ill try to have a single function for both of them!